core/txpool/validation.go

```diff
@@ -290,7 +290,7 @@ func ValidateTransactionWithState(tx *types.Transaction, signer types.Signer, op
 	}
 	cost := cost256.ToBig()
 	// we want to compare with effective gas balance here
-	cost = new(big.Int).Sub(cost, tx.Value())
+	balance = new(big.Int).Add(balance, tx.Value())
```

Comment:
It seems more intuitive to add tx.Value() back immediately after calling core.GetEffectiveGasBalance().

Reply:
Done in a0209bcfbac280949bee1424868864c3c6522146.